src/Form/AuthenticationForm.php

```diff
@@ -279,11 +278,15 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
     if (!empty($test_key->getKeyValue())) {
       /** @var \Drupal\apigee_edge\Plugin\KeyType\ApigeeAuthKeyType $test_key_type */
       $test_auth_type = $test_key->getKeyType()->getAuthenticationType($test_key);
-      if ($test_auth_type === EdgeKeyTypeInterface::EDGE_AUTH_TYPE_OAUTH) {
-        // Clear existing token data.
-        $this->oauthTokenStorage->removeToken();
-      }
       try {
+        if ($test_auth_type === EdgeKeyTypeInterface::EDGE_AUTH_TYPE_OAUTH) {
+          // Check the requirements first.
+          $this->oauthTokenStorage->checkRequirements();
+          // Clear existing token data.
+          $this->oauthTokenStorage->removeToken();
+        }
+        // TODO If email field contains an invalid email request should
```

Comment:
What is this comment about? We aren't sending an email here.

Reply:
Meh, not email request. The "Send request" button should not send an request to Apigee Edge if the form validation fails, like when the content of the email field is not an email